tock/employees/urls.py

```diff
@@ -1,11 +1,10 @@
-from django.conf.urls import url
+from django.urls import path
 
-from . import views
+from .views import UserListView, UserDetailView, UserFormView
 
+app_name="employees"
 urlpatterns = [
-    url(regex=r'^$', view=views.UserListView.as_view(), name='UserListView'),
-    url(regex=r'^(?P<username>[A-Za-z0-9._%+-]*)/$',
-        view=views.UserDetailView.as_view(), name='UserDetailView'),
-    url(regex=r'^e/(?P<username>[A-Za-z0-9._%+-]*)/$',
-        view=views.UserFormView.as_view(), name='UserFormView'),
+    path('', UserListView.as_view(), name='UserListView'),
+    path('<username>/', UserDetailView.as_view(), name='UserDetailView'),
+    path('e/<username>/', UserFormView.as_view(), name='UserFormView'),
 ]
```

Comment:
I see, I was worrying that we are allowing more additional characters than we needed to.  Just want to make sure that would be okay.

But sounds like it may not be a problem.

Reply:
That's a valid concern. We _are_ leaving username pretty open here, but they should be validated when the user is created, and if an invalid character comes in it'll just 404.